### ISSUE TITLE
Install failed with Cython 0.25

### ISSUE THREAD

**Issue by messense** [yaml/pyyaml#40]
Ref: https://github.com/cython/cython/issues/1498

https://github.com/cython/cython/pull/1456

**Comment by maartenbreddels** [comment]
Hit by the same issue, building on travis fails for osx:
https://s3.amazonaws.com/archive.travis-ci.org/jobs/170573675/log.txt
```
.....
      File "/Users/travis/build/maartenbreddels/vaex-wheels/venv/lib/python2.7/site-packages/Cython/Distutils/build_ext.py", line 19, in finalize_options

        self.distribution.ext_modules)

      File "/Users/travis/build/maartenbreddels/vaex-wheels/venv/lib/python2.7/site-packages/Cython/Build/Dependencies.py", line 809, in cythonize

        aliases=aliases)

      File "/Users/travis/build/maartenbreddels/vaex-wheels/venv/lib/python2.7/site-packages/Cython/Build/Dependencies.py", line 752, in create_extension_list

        **kwds))

    TypeError: __init__() takes exactly 6 arguments (3 given)
```
Using cython==0.24.1 seems to work.

**Comment by sigmavirus24** [comment]
I can't find the parameters that are required documented in Cython (`feature_name`, `feature_description`, `feature_check`). I won't have the time tonight to research this, but I hope to get to it before the weekend.

**Comment by maartenbreddels** [comment]
Would be great 👍

**Comment by robertwb** [comment]
These parameters are in yaml's custom Extension class. This was due to the switch to use cythonize(...) by default (which I'm planning on temporarily reverting).

**Comment by sigmavirus24** [comment]
@robertwb so, we're going to have problems when you switch to it anyway. Can you help me understand how I might go about fixing this? I'm not intimately familiar with cython or distutils. Thanks.

**Comment by robertwb** [comment]
Looks like you have quite the complicated setup.py file. This is going to take some thought, but I'll keep pyyaml in mind while rewriting https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#

**Comment by robertwb** [comment]
FYI, if you get travis ci up and running, it'd be easier for me to play with.

**Comment by sigmavirus24** [comment]
@robertwb Yeah, I've been trying to get a hold of someone who can add that. In the meantime, I assume we should be using something like https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#multiple-cython-files-in-a-package ?

**Comment by anirudt** [comment]
Is there any quick fix for the above? I got the same error while installing Keras on a Ubuntu 14.04 32 bit machine. Any help would be appreciated.

**Comment by sigmavirus24** [comment]
@robertwb we have Travis CI set up now.

**Comment by robertwb** [comment]
The key issue is that your Extension class

https://github.com/yaml/pyyaml/blob/16bd7d06c2cd4f2b688e2adc083b175595c03000/setup.py#L129

does not take the same arguments as distutils Extension class.

**Comment by anirudt** [comment]
I see, is this an easy fix, @sigmavirus24 ?
